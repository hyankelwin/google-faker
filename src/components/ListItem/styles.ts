import styled from 'styled-components';

export const Content = styled.div`
  padding-top: 30px;

  a {
    text-decoration: none;
  }
`;

export const Caption = styled.p`
  font-size: 12px;
`;
export const Title = styled.h3`
  padding-top: 5px;
  padding-bottom: 5px;
  cursor: pointer;
  color: #2c21b2;
`;
export const Description = styled.p`
  font-size: 14px;
`;
