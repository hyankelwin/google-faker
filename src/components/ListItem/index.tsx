import React from 'react';
import { Animal } from '../../api/google-faker-api/search/interfaces/animal.interface';
import { Content, Caption, Title, Description } from './styles';

interface ListItemProps {
  data: Animal[];
  onClickItem: (item: Animal) => void;
}

const ListItem: React.FC<ListItemProps> = ({ data, onClickItem }) => {
  return (
    <>
      {data.map(item => (
        <Content key={item.id}>
          <Caption>{item.site}</Caption>
          <Title data-testid="title-list" onClick={() => onClickItem(item)}>
            {item.name}
          </Title>
          <Description>{item.description}</Description>
        </Content>
      ))}
    </>
  );
};

export default ListItem;
