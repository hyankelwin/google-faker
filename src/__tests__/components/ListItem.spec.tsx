import { fireEvent, render } from '@testing-library/react';
import { mount } from 'enzyme';
import ListItem from '../../components/ListItem';
import { mockAnimalList } from '../../__mocks__/animal.mock';

describe('ListItem component', () => {
  describe('should be able to passed in all tests on index.tsx', () => {
    it('should be able to render ListItem component with success', () => {
      const props = {
        data: mockAnimalList,
        onClickItem: jest.fn(),
      };

      const wrapper = mount(<ListItem {...props} />);

      expect(wrapper.debug()).toMatchSnapshot();
    });

    it('should be able to simulate click on title of ListItem component', () => {
      const props = {
        data: mockAnimalList,
        onClickItem: jest.fn(),
      };

      const { getAllByTestId } = render(<ListItem {...props} />);

      const title = getAllByTestId('title-list')[0];

      fireEvent.click(title);

      expect(title).toBeTruthy();
    });
  });
});
