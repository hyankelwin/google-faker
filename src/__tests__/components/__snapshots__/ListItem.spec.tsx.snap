// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component should be able to passed in all tests on index.tsx should be able to render ListItem component with success 1`] = `
"<ListItem data={{...}} onClickItem={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }}>
  <styled.div>
    <div className=\\"sc-aXZVg iXJuVZ\\">
      <styled.p>
        <p className=\\"sc-gEvEer hyArYH\\">
          http://google.com.br
        </p>
      </styled.p>
      <styled.h3 data-testid=\\"title-list\\" onClick={[Function: onClick]}>
        <h3 data-testid=\\"title-list\\" onClick={[Function: onClick]} className=\\"sc-eqUAAy dmkfev\\">
          Siamês
        </h3>
      </styled.h3>
      <styled.p>
        <p className=\\"sc-fqkvVR cdqkYV\\">
          This cat is the most beautiful than all cats
        </p>
      </styled.p>
    </div>
  </styled.div>
  <styled.div>
    <div className=\\"sc-aXZVg iXJuVZ\\">
      <styled.p>
        <p className=\\"sc-gEvEer hyArYH\\">
          http://youtube.com.br
        </p>
      </styled.p>
      <styled.h3 data-testid=\\"title-list\\" onClick={[Function: onClick]}>
        <h3 data-testid=\\"title-list\\" onClick={[Function: onClick]} className=\\"sc-eqUAAy dmkfev\\">
          Maine Coon
        </h3>
      </styled.h3>
      <styled.p>
        <p className=\\"sc-fqkvVR cdqkYV\\">
          This cat is more furry than others
        </p>
      </styled.p>
    </div>
  </styled.div>
  <styled.div>
    <div className=\\"sc-aXZVg iXJuVZ\\">
      <styled.p>
        <p className=\\"sc-gEvEer hyArYH\\">
          http://uol.com.br
        </p>
      </styled.p>
      <styled.h3 data-testid=\\"title-list\\" onClick={[Function: onClick]}>
        <h3 data-testid=\\"title-list\\" onClick={[Function: onClick]} className=\\"sc-eqUAAy dmkfev\\">
          Angorá
        </h3>
      </styled.h3>
      <styled.p>
        <p className=\\"sc-fqkvVR cdqkYV\\">
          This cat is all white
        </p>
      </styled.p>
    </div>
  </styled.div>
</ListItem>"
`;
